packages/authorization-process/src/model/domain/apiConverter.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Client,
+  ClientKind,
+  Key,
+  KeyUse,
+  clientKind,
+  keyUse,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import {
+  ApiClient,
+  ApiClientWithKeys,
+  ApiClientKind,
+  ApiKey,
+  ApiKeyUse,
+} from "./models.js";
+
+export const ClientKindToApiClientKind = (kind: ClientKind): ApiClientKind =>
+  match<ClientKind, ApiClientKind>(kind)
+    .with(clientKind.consumer, () => "CONSUMER")
+    .with(clientKind.api, () => "API")
+    .exhaustive();
+
+export const KeyUseToApiKeyUse = (kid: KeyUse): ApiKeyUse =>
+  match<KeyUse, ApiKeyUse>(kid)
+    .with(keyUse.enc, () => "ENC")
+    .with(keyUse.sig, () => "SIG")
+    .exhaustive();
+
+export const clientToApiClient = (
+  client: Client,
+  withKeys: boolean
+): ApiClient | ApiClientWithKeys => ({
+  id: client.id,
+  name: client.name,
+  consumerId: client.consumerId,
+  users: client.users,
+  createdAt: client.createdAt.toJSON(),
+  purposes: client.purposes,
+  kind: ClientKindToApiClientKind(client.kind),
+  description: client.description ? client.description : undefined,
+  ...(withKeys ? { keys: client.keys } : {}),
+});
```

Comment:
I kinda dislike the "trailing boolean" in the arguments.

If we call this function, it would look like this:
```typescript
clientToApiClient(client, true)
```
At first sight it's not clear what that boolean does, to understand it we should hover over it or open the function, which is not a big deal, but it's a mental overhead we could easily avoid.

I would implement it like this:
```typescript
export const clientToApiClient = (
  client: Client,
 { includeKeys }: { includeKeys: boolean } 
): ApiClient | ApiClientWithKeys => ({ ... })
```

So that, when we call it, it is much more clear.

```typescript
clientToApiClient(client, { includeKeys: true })
```